rust_src/build.rs

```diff
@@ -594,7 +593,552 @@ fn generate_globals() {
+
+            builder = builder.clang_arg("-Demacs")
+                .header("wrapper.h")
+                .generate_inline_functions(true)
+                .derive_default(true)
+                .ctypes_prefix("::libc")
+                // we define these ourselves, for various reasons
+                .blacklist_type("Lisp_Object")
+                .blacklist_type("emacs_globals")
+                .blacklist_type("Q.*") // symbols like Qnil and so on
+                .blacklist_type("USE_LSB_TAG")
+                .blacklist_type("VALMASK")
+                .blacklist_type("PSEUDOVECTOR_FLAG")
+                // these two are found by bindgen on mac, but not linux
+                .blacklist_type("EMACS_INT_MAX")
+                .blacklist_type("VAL_MAX")
+                //.blacklist_type("staticpro")
+                // this is wallpaper for a bug in bindgen, we don't lose much by it
+                // https://github.com/servo/rust-bindgen/issues/687
+                .blacklist_type("BOOL_VECTOR_BITS_PER_CHAR")
+                // this is wallpaper for a function argument that shadows a static of the same name
+                // https://github.com/servo/rust-bindgen/issues/804
+                .blacklist_type("face_change")
+                // these never return, and bindgen doesn't yet detect that, so we will do them manually
+                .blacklist_type("error")
+                .blacklist_type("circular_list")
+                .blacklist_type("wrong_type_argument")
+                .blacklist_type("nsberror")
+                .blacklist_type("emacs_abort")
+                .blacklist_type("Fsignal")
+                .blacklist_type("memory_full")
+                .blacklist_type("bitch_at_user")
+                .blacklist_type("wrong_choice")
+                .blacklist_type("wrong_range")
+                // these are defined in data.rs
+                .blacklist_type("Lisp_Fwd")
+                .blacklist_type("Lisp_.*fwd")
+                // these are defined in remacs_lib
+                .blacklist_type("timespec")
+                .blacklist_type("current_timespec")
+                .blacklist_type("timex")
+                .blacklist_type("clock_adjtime")
+                // bindgen fails to generate this one correctly; it's hard
+                // https://github.com/rust-lang-nursery/rust-bindgen/issues/1318
+                .blacklist_type("max_align_t")
+                // these enums are better as simple constants
+                .constified_enum("EMACS_INT_WIDTH")
+                .constified_enum("BOOL_VECTOR_BITS_PER_CHAR")
+                .constified_enum("BITS_PER_BITS_WORD")
+                // these enums all meet the requirements to also be Rust enums (in theory). 90% of
+                // these are present in struct fields, return values, etc; we don't have much
+                // choice but to inspect each one or just hope that they're correct; guess which
+                // I've done?
```

Comment:
Could you be a little more definitive here? Something like "these seem to be sensible, but I haven't closely checked them" perhaps.